staging_service/autodetect/GenerateMappings.py

```diff
@@ -0,0 +1,262 @@
+]
+
+mapping[ZIP] = [
+    {
+        "title": "Decompress/Unpack",
+        "app": "kb_uploadmethods/unpack_staging_file",
+        "output_type": [None],
+    }
+]
+mapping[CSV] = [
+    {
+        "title": "Samples",
+        "app": "sample_uploader/import_samples",
+        "output_type": ["KBaseSets.SampleSet"],
+    }
+]
+
+mapping[TSV] = [
+    {
+        "title": "Media",
+        "app": "kb_uploadmethods/import_tsv_excel_as_media_from_staging",
+        "output_type": ["KBaseBiochem.Media"],
+    },
+    # {
+    #     "title": "Attribute Mapping",
+    #     "app": "kb_uploadmethods/import_attribute_mapping_from_staging",
+    #     "output_type": ["KBaseExperiments.AttributeMapping"],
+    # },
+    {
+        "title": "Expression Matrix",
+        "app": "kb_uploadmethods/import_tsv_as_expression_matrix_from_staging",
+        "output_type": ["KBaseFeatureValues.ExpressionMatrix"],
+    },
+    {
+        "title": "Metabolic Annotations",
+        "app": "MergeMetabolicAnnotations/import_annotations",
+        "output_type": ["KBaseGenomes.Genome"],
+    },
+    {
+        "title": "Bulk Metabolic Annotations",
+        "app": "MergeMetabolicAnnotations/import_bulk_annotations",
+        "output_type": ["KBaseGenomes.Genome"],
+    },
+    {
+        "title": "FBA Model",
+        "app": "kb_uploadmethods/import_file_as_fba_model_from_staging",
+        "output_type": ["KBaseFBA.FBAModel"],
+    },
+    {
+        "title": "Phenotype Set",
+        "app": "kb_uploadmethods/import_tsv_as_phenotype_set_from_staging",
+        "output_type": ["KBasePhenotypes.PhenotypeSet"],
+    },
+]
+
+mapping[EXCEL] = [
+    {
+        "title": "Samples",
+        "app": "sample_uploader/import_samples",
+        "output_type": ["KBaseSets.SampleSet"],
+    },
+    {
+        "title": "Media",
+        "app": "kb_uploadmethods/import_tsv_excel_as_media_from_staging",
+        "output_type": ["KBaseBiochem.Media"],
+    },
+    {
+        "title": "Attribute Mapping",
+        "app": "kb_uploadmethods/import_attribute_mapping_from_staging",
+        "output_type": ["KBaseExperiments.AttributeMapping"],
+    },
+    {
+        "title": "FBA Model",
+        "app": "kb_uploadmethods/import_file_as_fba_model_from_staging",
+        "output_type": ["KBaseFBA.FBAModel"],
+    },
+]
+mapping[JSON] = [
+    {
+        "title": "EscherMap",
+        "app": "kb_uploadmethods/import_eschermap_from_staging",
+        "output_type": ["KBaseFBA.EscherMap"],
+    }
+]
+
+mapping[SBML] = [
+    {
+        "title": "FBA Model",
+        "app": "kb_uploadmethods/import_file_as_fba_model_from_staging",
+        "output_type": ["KBaseFBA.FBAModel"],
+    }
+]
+
+new_apps = OrderedDict()
+extensions_flat = []
+
+counter = 0
+for category in mapping:
+    apps = mapping[category]
+    for app in apps:
+        # print("looking at", app)
+        title = app["title"]
+        if title not in new_apps:
+            app["extensions"] = []
+            app["id"] = counter
```

Comment:
To make it more readable, `id` could be title lowercased with punctuation and spaces replaced by `_`.

Reply:
@ialarmedalien Thats true, but for the case of many files with many matches to apps (especially once we start giving non 0/1 weights),  the list size in the response could quickly grow in size (but this might be a premature optimization to return int ids instead of strings)

Reply:
I would switch it over to numeric IDs if/when the response gets too large, and should we start running into problems with response size, we can also think compressing the response. Clarity > brevity at this stage, though.

Reply:
I switched it over already a bit further down